shell/platform/linux/fl_keyboard_manager_test.cc

```diff
@@ -432,61 +422,59 @@ class KeyboardTester {
   }
 
   void respondToEmbedderCallsWith(bool response) {
-    fl_mock_view_set_embedder_handler(
-        view_, [response, this](const FlutterKeyEvent* event,
-                                const AsyncKeyCallback& callback) {
-          EXPECT_FALSE(during_redispatch_);
-          callback(response);
-        });
+    embedder_handler_ = [response, this](const FlutterKeyEvent* event,
+                                         const AsyncKeyCallback& callback) {
+      EXPECT_FALSE(during_redispatch_);
+      callback(response);
+    };
   }
 
   void recordEmbedderCallsTo(std::vector<CallRecord>& storage) {
-    fl_mock_view_set_embedder_handler(
-        view_, [&storage, this](const FlutterKeyEvent* event,
-                                AsyncKeyCallback callback) {
-          EXPECT_FALSE(during_redispatch_);
-          auto new_event = std::make_unique<FlutterKeyEvent>(*event);
-          char* new_event_character = cloneString(event->character);
-          new_event->character = new_event_character;
-          storage.push_back(CallRecord{
-              .type = CallRecord::kKeyCallEmbedder,
-              .callback = std::move(callback),
-              .event = std::move(new_event),
-              .event_character = std::unique_ptr<char[]>(new_event_character),
-          });
-        });
+    embedder_handler_ = [&storage, this](const FlutterKeyEvent* event,
+                                         AsyncKeyCallback callback) {
+      EXPECT_FALSE(during_redispatch_);
+      auto new_event = std::make_unique<FlutterKeyEvent>(*event);
+      char* new_event_character = cloneString(event->character);
+      new_event->character = new_event_character;
```

Comment:
Maybe for a followup PR given that this is just refactoring existing code, but I wonder if a static helper method like this might be helpful:
```
std::unique_ptr<FlutterKeyEvent> CloneEvent(const FlutterKeyEvent* event);
```
I doubt we'll add more heap-allocated fields to FlutterKeyEvent, but it might help readability by declaring the overall intent (cloning the event) rather than step-by-step cloning it in a bunch of these.

Reply:
I think a follow up PR. I'm not sure how much of this code will survive by the end of the refactoring.